src/item_group.cpp

```diff
@@ -345,7 +345,7 @@ Item_spawn_data::ItemList Item_group::create(int birthday, RecursionList &rec) c
 
     for( auto& e : result ) {
         if( e.is_tool() || e.is_gun() || e.is_magazine() ) {
-            bool spawn_ammo = rng( 0, 99 ) < with_ammo && e.ammo_remaining() == 0;
+            bool spawn_ammo = rng( 0, 99 ) < with_ammo && e.ammo_remaining() == 0 && e.ammo_type() != "NULL";
             bool spawn_mag  = rng( 0, 99 ) < with_magazine && !e.magazine_integral() && !e.magazine_current();
 
             if( spawn_mag || spawn_ammo ) {
```

Comment:
This looks to be the more suspicious line. An item using an integral magazine doesn't have a default magazine...